pkg/action/bootentries.go

```diff
@@ -229,11 +231,28 @@ func bootNameToSystemdConf(name string) (string, error) {
 // and prompts the user to select one
 // then calls the underlying SelectBootEntry function to mange the entry writing and validation
 func listBootEntriesSystemd(cfg *config.Config) error {
+	var err error
+	e := elemental.NewElemental(cfg)
+	cleanup := utils.NewCleanStack()
+	defer func() { err = cleanup.Cleanup(err) }()
 	// Get EFI partition
 	efiPartition, err := partitions.GetEfiPartition()
 	if err != nil {
 		return err
 	}
+	// mount if not mounted
+	if mounted, _ := utils.IsMounted(cfg, efiPartition); !mounted {
```

Comment:
we should better not skip the error here. After all, we are checking for `false` which could be returned in case of an error, thus entering the block when things don't even work.

This methods already has error handling so let's handle this one too.

Reply:
good point, done!